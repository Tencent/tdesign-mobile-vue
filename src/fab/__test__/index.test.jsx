import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import Fab from '../fab.vue';
import { AppIcon as TIconApp, MoreIcon } from 'tdesign-icons-vue-next';

describe('fab', () => {
  describe('props', () => {
    it(': icon is a function', async () => {
      const icon = () => <TIconApp />;
      const wrapper = mount(Fab, {
        props: { icon },
      });
      expect(wrapper.findComponent(TIconApp).exists()).toBeTruthy();
    });

    it(': text', async () => {
      const text = 'fab';
      const wrapper = mount(Fab, {
        props: { text },
      });
      const textContainer = wrapper.find('.t-fab__text');
      expect(textContainer.text()).toBe(text);
    });

    it(': style', () => {
      const style = 'right: 32px; bottom: 48px;';
      const wrapper = mount(Fab, {
        props: { style },
      });
      const buttonContainer = wrapper.find('.t-button');
      expect(getComputedStyle(wrapper.element, null).right).toBe('32px');
      expect(getComputedStyle(wrapper.element, null).bottom).toBe('48px');
    });

    it(': buttonProps', () => {
      const icon = () => <TIconApp />;
      const buttonProps = {
        size: 'large',
        shape: 'circle',
        theme: 'danger',
      };
      const wrapper = mount(Fab, {
        props: { icon, buttonProps },
      });
      expect(wrapper.classes()).toContain(`t-button--danger`);
      expect(wrapper.classes()).toContain(`t-button--circle`);
      expect(wrapper.classes()).toContain(`t-button--size-large`);
    });
  });

  describe('slots', () => {
    it(': icon', () => {
      const icon = () => <TIconApp />;
      const wrapper = mount(Fab, {
        slots: {
          icon,
        },
      });
      expect(wrapper.findComponent(TIconApp).exists()).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('event', () => {
    it(': click', async () => {
      const onClick = vi.fn();
      const wrapper = mount({
        render() {
          return <Fab onClick={onClick} />;
        },
      });
      // 触发事件的节点
      await wrapper.trigger('click');
      // 判断事件是否被触发
      expect(onClick).toBeCalledTimes(1);
    });
  });
});
