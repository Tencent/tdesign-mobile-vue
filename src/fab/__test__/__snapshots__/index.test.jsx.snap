// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fab > slots > : icon 1`] = `
<div
  class="t-fab"
  style="right: 16px; bottom: 32px;"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--base t-button--primary t-button--circle t-fab__button"
    role="button"
    type="button"
  >
    
    <svg
      class="t-icon t-icon-app"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M2 2h9v9H2V2zm2 2v5h5V4H4zm13.5 0a2.5 2.5 0 100 5 2.5 2.5 0 000-5zM13 6.5a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM2 13h9v9H2v-9zm2 2v5h5v-5H4zm9-2h9v9h-9v-9zm2 2v5h5v-5h-5z"
        fill="currentColor"
      />
    </svg>
    
    <span
      class="t-button__content"
    >
      
      
      
    </span>
    <!---->
  </button>
</div>
`;
